ethcore/src/builtin.rs

```diff
@@ -309,7 +309,7 @@ impl Impl for ModexpImpl {
 			base = base % &modulus;
 
 			// fast path for base divisible by modulus.
-			if base.is_zero() { return result }
+			if base.is_zero() { return BigUint::zero() }
```

Comment:
isn't this done here?

https://github.com/paritytech/parity/pull/6424/files#diff-f01cea79b5ea6391c260249a4cf801a5R303

and tested here

https://github.com/paritytech/parity/pull/6424/files#diff-f01cea79b5ea6391c260249a4cf801a5L660

Reply:
No. the place you mentioned is the original base. But here the base is calculated via `base % &modulus`. See: 

https://github.com/paritytech/parity/pull/6424/files#diff-f01cea79b5ea6391c260249a4cf801a5R309

Reply:
Ah, right. A test would be good

Reply:
But I think we can just move `base = base % &modulus;` up to the first line in this function. and remove the test here. All the condition will be tested at:

https://github.com/paritytech/parity/pull/6424/files#diff-f01cea79b5ea6391c260249a4cf801a5R303

WDYT? @rphmeier @debris

Reply:
Then it would panic if the modulo was zero, which is protected at the given point.

Reply:
Ah, right. 👍

Reply:
I am struggling for the test. There are more stuffs besides the `modexp` function. I can easily test the `modexp` function, however since it is just a inner function, I have to test the `ModexpImpl::execute`. I will be really appreciated if there are some instructions.

Reply:
oh just move the function outside to the mod level and add a test, there are plenty of support functions there already